fundingmanager_test.go

```diff
@@ -2384,6 +2392,126 @@ func TestFundingManagerPrivateRestart(t *testing.T) {
 	assertNoChannelState(t, alice, bob, fundingOutPoint)
 }
 
+func TestFundingManagerCustomChannelAcceptor(t *testing.T) {
+	t.Parallel()
+
+	// ChannelAcceptor that rejects some open channel requests.
+	acceptor := mockChannelAcceptor{
+		acceptClosure: func(req *chanacceptor.ChannelAcceptRequest) error {
+			// A rejection error without message. Default error message is
+			// expected to be sent in such case.
+			if req.OpenChanMsg.PushAmount < lnwire.NewMSatFromSatoshis(5) {
+				return errors.New("")
+			}
+
+			// A rejection error with custom message. This message is expected
+			// to be sent to the remote node.
+			if req.OpenChanMsg.PushAmount < lnwire.NewMSatFromSatoshis(10) {
+				return errors.New("push amount too small")
+			}
+
+			return nil
+		},
+	}
+
+	// Set the rejecting channel acceptor as the open channel predicate.
+	alice, bob := setupFundingManagers(
+		t, func(cfg *fundingConfig) {
+			cfg.OpenChannelPredicate = acceptor
+		},
+	)
+	defer tearDownFundingManagers(t, alice, bob)
+
+	var initReqs []*openChanReq
+	for _, pushAmt := range []lnwire.MilliSatoshi{
+		lnwire.NewMSatFromSatoshis(1),
+		lnwire.NewMSatFromSatoshis(6),
+		lnwire.NewMSatFromSatoshis(11),
+	} {
+		updateChan := make(chan *lnrpc.OpenStatusUpdate)
+		errChan := make(chan error, 1)
+		initReq := &openChanReq{
+			targetPubkey:    bob.privKey.PubKey(),
+			chainHash:       *activeNetParams.GenesisHash,
+			localFundingAmt: 500000,
+			pushAmt:         pushAmt,
+			private:         true,
+			updates:         updateChan,
+			err:             errChan,
+		}
+		initReqs = append(initReqs, initReq)
+	}
+
+	var rejected []error
+	var accepted []*lnwire.AcceptChannel
+	for i, initReq := range initReqs {
+		// Create a funding request and start the workflow.
+		alice.fundingMgr.initFundingWorkflow(bob, initReq)
+
+		// Alice should have sent the OpenChannel message to Bob.
+		var aliceMsg lnwire.Message
+		select {
+		case aliceMsg = <-alice.msgChan:
+		case err := <-initReq.err:
+			t.Fatalf("error init funding workflow: %v", err)
+		case <-time.After(time.Second * 5):
+			t.Fatalf("alice did not send OpenChannel message")
+		}
+
+		openChannelReq, ok := aliceMsg.(*lnwire.OpenChannel)
+		if !ok {
+			errorMsg, gotError := aliceMsg.(*lnwire.Error)
+			if gotError {
+				t.Fatalf("expected OpenChannel to be sent "+
+					"from bob, instead got error: %v",
+					errorMsg.Error())
+			}
+			t.Fatalf("expected OpenChannel to be sent from "+
+				"alice, instead got %T", aliceMsg)
+		}
+
+		// Let Bob handle the init message.
+		bob.fundingMgr.processFundingOpen(openChannelReq, alice)
+
+		// Assert Bob responded.
+		// First two requests should be rejected.
+		if i < 2 {
```

Comment:
this can just be a bool that is unrelated to the position in the array. A field in the test called `expectBobRespond` for example